### PR TITLE
exception class changed. fixes #447

### LINKED ISSUE #447: sshkey upload fails with odd error message
I can generate a key and save it with `travis sshkey --generate`. However, when I attempt to reuse this key with another repository, I always seem to get this error message:

```
$ travis sshkey --upload ~/.ssh/id_travis_rsa"
password must be shorter than 1024 bytes
for a full error report, run travis report --pro

$ travis report --pro
System
Ruby:                     Ruby 2.4.0-p-1
Operating System:         Mac OS X 10.12
RubyGems:                 RubyGems 2.6.7

CLI
Version:                  1.8.2
Plugins:                  none
Auto-Completion:          yes
Last Version Check:       2016-10-20 17:29:15 -0500

Session
API Endpoint:             https://api.travis-ci.com/
Logged In:                as "westhomas"
Verify SSL:               yes
Enterprise:               no

Endpoints
org:                      https://api.travis-ci.org/ (access token)
pro:                      https://api.travis-ci.com/ (access token, current)

Last Exception
An error occurred running `travis sshkey --pro`:
    OpenSSL::OpenSSLError: password must be shorter than 1024 bytes
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/tools/ssl_key.rb:21:in `initialize'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/tools/ssl_key.rb:21:in `new'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/tools/ssl_key.rb:21:in `has_passphrase?'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/cli/sshkey.rb:79:in `remove_passphrase'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/cli/sshkey.rb:38:in `update_key'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/cli/sshkey.rb:22:in `run'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/cli/command.rb:198:in `execute'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/lib/travis/cli.rb:64:in `run'
        from /Users/wes/.rvm/gems/ruby-head/gems/travis-1.8.2/bin/travis:18:in `<top (required)>'
        from /Users/wes/.rvm/gems/ruby-head/bin/travis:22:in `load'
        from /Users/wes/.rvm/gems/ruby-head/bin/travis:22:in `<main>'
        from /Users/wes/.rvm/gems/ruby-head/bin/ruby_executable_hooks:15:in `eval'
        from /Users/wes/.rvm/gems/ruby-head/bin/ruby_executable_hooks:15:in `<main>'
```

I can simply generate a new key for each repo, but this seems wrong.

### PR DESCRIPTION
**Pull request by rmoriz**
see #447

### CODE REVIEW

File: lib/travis/tools/ssl_key.rb [hunk 1]
```diff
@@ -20,7 +20,9 @@ def public_rsa_key(string)
       def has_passphrase?(key)
         OpenSSL::PKey::RSA.new(key, key)
         false
-      rescue OpenSSL::PKey::RSAError
+      rescue OpenSSL::PKey::RSAError, OpenSSL::OpenSSLError
```
Comment (BanzaiMan) [inline lib/travis/tools/ssl_key.rb]: `OpenSSL::PKey::RSAError` is a subclass of `OpenSSL::OpenSSLError`, so there is no need to `rescue` the former.

**Comment by alper** [comment]
@rmoriz Could you respond to the comment?

**Comment by rmoriz** [comment]
@alper I don't use travis anymore (see date of PR). "Allow edits from maintainers" is on, so feel free to change PR.

**Comment by micahstubbs** [comment]
@alper could you edit the PR and do what it takes to get this in? lots of us over on https://github.com/travis-ci/travis.rb/issues/447 are waiting for the fix 😅

**Comment by micahstubbs** [comment]
maybe @igorwwwwwwwwwwwwwwwwwwww can help make the changes?

**Comment by alper** [comment]
@micahstubbs I am not a part of this org. I had a similar issue and gave up.

**Comment by micahstubbs** [comment]
huh, I see.   maybe tagging @travis-ci will help?

**Comment by micahstubbs** [comment]
ah so it's @BanzaiMan who's help we need 💡 

@BanzaiMan  could you edit the PR and do what it takes to get this in? lots of us over on #447 are waiting for the fix 😅

**Comment by alper** [comment]
@micahstubbs I don't know. If they haven't managed to push this near trivial fix in the past year, I don't think they'll manage to do it now.

**Comment by dyejon** [comment]
fix works for me .. fwiw

**Comment by rogerluan** [comment]
Can someone merge this in? It's a very annoying issue having to generate a new key for every repo 😕

**Comment by thesciz** [comment]
@BanzaiMan @alper - this can't be that much effort to get in; I hate having to downgrade Ruby versions just to avoid this. Can we get a little mergey mergey?

**Comment by BanzaiMan** [comment]
Resolved by https://github.com/travis-ci/travis.rb/pull/700.

**Pull request closed without merging**